lms/djangoapps/courseware/tests/test_module_render.py

```diff
@@ -607,72 +608,75 @@ def test_score_without_student_state(self):
 
     @XBlock.register_temp_plugin(StubCompletableXBlock, identifier='comp')
     def test_completion_event(self):
-        course = CourseFactory.create()
-        block = ItemFactory.create(category='comp', parent=course)
-        request = self.request_factory.post(
-            '/',
-            data=json.dumps({'completion': 0.625}),
-            content_type='application/json',
-        )
-        request.user = self.mock_user
-        response = render.handle_xblock_callback(
-            request,
-            unicode(course.id),
-            quote_slashes(unicode(block.scope_ids.usage_id)),
-            'complete',
-            '',
-        )
-        self.assertEqual(response.status_code, 200)
-        completion = BlockCompletion.objects.get(block_key=block.scope_ids.usage_id)
-        self.assertEqual(completion.completion, 0.625)
+        with completion_waffle.waffle().override(completion_waffle.ENABLE_COMPLETION_TRACKING, True):
```

Comment:
Nit: I'd prefer to have this override as a method decorator, like you're already doing for `XBlock.register_temp_plugin` - I believe waffle's `override` method supports that, and then the indentation is nicer though. Consider this totally optional!

Reply:
It works for waffle directly, but not the edX waffle wrapper. (Until python 3.2, which gives that behavior for free with the contextlib.contextmanager decorator). Let's leave it for now.

Reply:
I thought there was a decorator?  I know there is one for our Waffle Flags. :)